### PR TITLE
Update theme thank you pages based on paid/free status of the purchase

### PR DESCRIPTION
**Pull request by lsl**
Fixes https://github.com/Automattic/dotcom-forge/issues/2719

## Proposed Changes

* Title
* Subtitle
* Footer text
* Back button
* Misc css
* Section layout

## Testing Instructions

* http://calypso.localhost:3000/marketplace/thank-you/test77753.wordpress.com?themes=muscat
* Compare figma VLoPHWqcewxNKZYjjkDu3O-fi-2414_3286

Before

Screenshot 2023-06-21 at 16-16-06 Test Business Simple Site — WordPress com

After

Free

Screenshot 2023-06-22 at 15-17-31 Storia Theme

Paid

Screenshot 2023-06-22 at 15-11-04 Atomic Test Site — WordPress com

Premium (broken)

Screenshot 2023-06-22 at 15-19-52 Storia Theme

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/marketplace/use-themes-thank-you-data.tsx [hunk 1]
```diff
@@ -90,12 +84,29 @@ export function useThemesThankYouData( themeSlugs: string[] ): ThankYouData {
 	);
 	const isAtomicNeeded = hasDotOrgThemes || hasExternallyManagedThemes;
 
+	const hasFreeThemes = themesList.some( ( theme ) => theme?.cost?.number === 0 );
```
Comment (lsl) [inline client/my-sites/checkout/checkout-thank-you/marketplace/use-themes-thank-you-data.tsx]: Is there a better way to do this or will this do?

Comment (candy02058912) [reply client/my-sites/checkout/checkout-thank-you/marketplace/use-themes-thank-you-data.tsx]: This is returning `false` when there is a free theme selected, I tried it out with the free theme storia 🤔 .

Comment (candy02058912) [reply client/my-sites/checkout/checkout-thank-you/marketplace/use-themes-thank-you-data.tsx]: I tried using `canUseTheme` to check instead. Could you also help test it out to see if it works as intended?

File: client/my-sites/checkout/checkout-thank-you/marketplace/use-themes-thank-you-data.tsx [hunk 2]
```diff
@@ -84,7 +84,13 @@ export function useThemesThankYouData( themeSlugs: string[] ): ThankYouData {
 	);
 	const isAtomicNeeded = hasDotOrgThemes || hasExternallyManagedThemes;
 
-	const hasFreeThemes = themesList.some( ( theme ) => theme?.cost?.number === 0 );
+	// theme will automatically activate if user has a plan that includes it
+	// the CTA is conditional to whether a theme activates automatically or not
+	const hasThemeIncludedInCurrentPlan = useSelector( ( state ) =>
+		allThemesFetched
```
Comment (lsl) [inline client/my-sites/checkout/checkout-thank-you/marketplace/use-themes-thank-you-data.tsx]: Nice catch

File: client/my-sites/checkout/checkout-thank-you/marketplace/use-themes-thank-you-data.tsx [hunk 3]
```diff
@@ -90,12 +84,35 @@ export function useThemesThankYouData( themeSlugs: string[] ): ThankYouData {
 	);
 	const isAtomicNeeded = hasDotOrgThemes || hasExternallyManagedThemes;
 
+	// theme will automatically activate if user has a plan that includes it
+	// the CTA is conditional to whether a theme activates automatically or not
+	const hasThemeIncludedInCurrentPlan = useSelector( ( state ) =>
+		allThemesFetched
+			? themesList.map( ( theme ) => siteId && canUseTheme( state, siteId, theme.id ) )
```
Comment (lsl) [inline client/my-sites/checkout/checkout-thank-you/marketplace/use-themes-thank-you-data.tsx]:
```suggestion
			? themesList.some( ( theme ) => siteId && canUseTheme( state, siteId, theme.id ) )
```

Should we use some here?

**Comment by lsl** [comment]
@candy02058912 

> I tried using canUseTheme to check instead. Could you also help test it out to see if it works as intended?

Don't think so, switched to 
```
	const hasPaidThemes = themesList.some( ( theme ) => theme?.product_details || theme?.price );
```

`product_details` covers paid themes, `price` covers premium themes. (apparently we have different data formats depending on theme type.)

What I really would like to know is where I got cost.number from `theme?.cost?.number === 0` I definitely saw `theme.cost: {number: 0}` somewhere.. not sure where though.

**Edit: hang on, now I see what you're talking about re activate cta. This won't work...**

**Review by lupus2k — COMMENTED** [review]
All the design looks great and follow all the Figma margins, gaps, etc. 
image
image
image

**Pull request closed without merging**